### PR TITLE
knative: add custom loader for sources

### PR DESCRIPTION
**Pull request by lburgazzoli**
This new loader enrich terminates a route with an uri like:

    knative://endpoint/{{env:KNATIVE_SINK:synk}}

It does not support sources that define more than one route.

### CODE REVIEW

File: camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                //TODO: this is a little hack as then configureRoutes will
+                //      be invoked twice: 1 by this hack and 1 by delegated
+                //      builder. Maybe, we should add builder lifecycle events.
+                List<RouteDefinition> definitions = builder.configureRoutes(context).getRoutes();
+
+                if (definitions.size() > 1) {
+                    throw new IllegalArgumentException("Cannot determine route top enrich");
+                }
+
+                final String sink = context.resolvePropertyPlaceholders("{{env:KNATIVE_SYNC:sink}}");
```
Comment (nicolaferraro) [inline camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java]: Maybe just call it "SINK", not "SYNC"

Comment (lburgazzoli) [reply camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java]: Beside the spelling, do you want to be named `SYNK` without `KNATIVE_` prefix ?

File: camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+            @Override
+            public void addRoutesToCamelContext(CamelContext context) throws Exception {
+                //TODO: this is a little hack as then configureRoutes will
+                //      be invoked twice: 1 by this hack and 1 by delegated
+                //      builder. Maybe, we should add builder lifecycle events.
+                List<RouteDefinition> definitions = builder.configureRoutes(context).getRoutes();
+
+                if (definitions.size() > 1) {
+                    throw new IllegalArgumentException("Cannot determine route top enrich");
```
Comment (nicolaferraro) [inline camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java]: A better approach would be to just WARN here. Translated to the KnativeSource context, it means that when the user declares multiple routes (in camel k mode), he nneds to define the output endpoint explicitly.

File: camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java (hunk 1 above)
Comment (nicolaferraro) [reply camel-k-loader-knative/src/main/java/org/apache/camel/k/loader/knative/KnativeSourceRoutesLoader.java]: No `KNATIVE_SINK` is ok ;)

**Pull request merged**